File: Data Structures & Algorithms/trapping-rain-water/submission-0.cs
 public class Solution
 {
     public int Trap(int[] height)
     {
         int[] prefix = new int[height.Length];
         prefix[0] = height[0];
         for (int i = 1; i < height.Length; i++)
         {
             prefix[i] = Math.Max(height[i], prefix[i - 1]);
         }
         int[] suffix = new int[height.Length];
         suffix[height.Length -1] = height[height.Length - 1];
         for(int i = height.Length -2; i>=0; i--)
         {
             suffix[i] = Math.Max(height[i],  suffix[i + 1]);
         }
         int sum = 0;
         for (int i = 0; i < height.Length; i++)
         {
             sum += Math.Min(prefix[i],suffix[i]) - height[i];
         }
         return sum;
     }
 }

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights)
    {
        int l = 0, r = heights.length -1;
        int max = Math.min(heights[l], heights[r]) * (r-l);
        while(l < r)
        {
            if(heights[l] > heights[r])
                r--;
            else
                l++;
            int tmp = Math.min(heights[l], heights[r]) * (r-l);

            max = Math.max(tmp, max);
        }
        return max;
    }
}